lib/data/wrapper.js

```diff
@@ -24,20 +26,27 @@ if (config.backends.data === 'mem') {
 }
 
 const data = {
-    put: (value, keyContext, log, cb) => {
+    put: (value, valueSize, keyContext, log, cb) => {
+        assert.strictEqual(typeof valueSize, 'number');
         log.debug('sending put to datastore', { implName, keyContext,
-            method: 'put' });
-        client.put(value, keyContext, log.getSerializedUids(), (err, key) => {
-            if (err) {
-                log.error('error from datastore', { error: err, implName });
-                return cb(errors.InternalError);
-            }
-            const dataRetrievalInfo = {
-                key,
-                dataStoreName: implName,
-            };
-            return cb(null, dataRetrievalInfo);
+                                                method: 'put' });
+        const hashedStream = new MD5Sum((err, hash) => {
+            value.calculatedHash = hash;
         });
+        value.pipe(hashedStream);
+        client.put(hashedStream, valueSize, keyContext, log.getSerializedUids(),
```

Comment:
This call not needed to be done after the md5 calculation, in the done callback ? :)